QuestionAnswering/QA_ICL/backend/triton_inference_server/README.md

```diff
@@ -1,27 +1,32 @@
 # Self-hosting ML models for QA applications
```

Comment:
A general comment is that I have no idea what this module is used for? Like does it serve to generate some data outputs and must the agent continuously interact with the container?

If no code is required and this directory is to show directory structure, I might suggest to document that, and potentially remove all the instructions and move it elsewhere. 

In general the documentation here is confusing and does not have any steps for using it, or does not explain that it does not have any steps

Reply:
> I have no idea what this module is used for?

This directory is to spin up the Triton Inference Server that serves the Sentence-BERT model for generating text embedding. This is mentioned in the file `QA_ICL/README.md`.

> does it serve to generate some data outputs and must the agent continuously interact with the container?

The FastAPI app relies this service to obtain text embeddings. The FastAPI app interacts with this service for the data ingestion process and for every request submitted to the endpoint `/qa`.

> If no code is required and this directory is to show directory structure, I might suggest to document that, and potentially remove all the instructions and move it elsewhere.

It is possible to write Python codes for Triton Inference Server (see [QuestionAnswering/QA_SEQ2SEQ/deployment/triton/model_repository/feature_extraction/1/model.py](https://github.com/cambridge-cares/TheWorldAvatar/blob/main/QuestionAnswering/QA_SEQ2SEQ/deployment/triton/model_repository/feature_extraction/1/model.py)). The current directory contains only the config file `backend/triton_inference_server/model_repository/mpnet/config.pbtxt`, which helps to configure the Sentence-BERT model weights to be placed in `backend/triton_inference_server/model_repository/mpnet/1/`. It is possible that in the future to serve a different model, some additional code will be added.

> the documentation here is confusing and does not have any steps for using it, or does not explain that it does not have any steps

There are sections "Installation" and "Steps" that outline how to spin up this service. I have updated the steps to include the initial action of depositing the model weights in the relevant directory.

Reply:
> This directory is to spin up the Triton Inference Server that serves the Sentence-BERT model for generating text embedding. This is mentioned in the file QA_ICL/README.md.

Add this answer to the first line. This explains much more than the current first line

> The FastAPI app relies this service to obtain text embeddings. The FastAPI app interacts with this service for the data ingestion process and for every request submitted to the endpoint /qa.

Add this answer to the intro as well. It is useful info

I have misunderstood the customisation required for the triton server, sorry about that!  But do mention that the current FastAPI app requires an additional config file to the default Triton docker image or whatever this is supposed to be. Also add the `/1/` directory with a `.gitignore` like the below and commit it so that users know to put it here
```
# Ignore everything in this directory
*
# Except this file
!.gitignore
```

> There are sections "Installation" and "Steps" that outline how to spin up this service. I have updated the steps to include the initial action of depositing the model weights in the relevant directory.

I think remove the reference and edit the sentence to include the file path [model_repository/mpnet/1](https://github.com/cambridge-cares/TheWorldAvatar/main/QuestionAnswering/QA_ICL/backend/triton_inference_server/model_repository/mpnet/1).  It is a little tedious to keep bouncing between pages/sections for one liners

Please ensure the file path points to main rather than the current branch so that it works after merge.

Reply:
Thank you for your detailed feedback. I've moved detailed resource information to the Steps section. Added instructions for placing resource files

Reply:
I've made improvements to address your review:

 - Clarified the module's purpose

 - Added a /1/ directory under model_repository/mpnet/ with a .gitignore file to indicate where users should put the model weights and prevent accidental commits.